src/core/ddsi/include/dds/ddsi/ddsi_typewrap.h

```diff
@@ -29,6 +29,11 @@ extern "C" {
 
 #define XT_FLAG_EXTENSIBILITY_MASK  0x7
 
+#define DDS_XTypes_TRY_CONSTRUCT_INVALID 0
+#define DDS_XTypes_TRY_CONSTRUCT_DISCARD DDS_XTypes_TRY_CONSTRUCT1
+#define DDS_XTypes_TRY_CONSTRUCT_USE_DEFAULT DDS_XTypes_TRY_CONSTRUCT2
+#define DDS_XTypes_TRY_CONSTRUCT_TRIM DDS_XTypes_TRY_CONSTRUCT1 | DDS_XTypes_TRY_CONSTRUCT2
```

Comment:
Defining the macro like this is really asking for trouble ...

```suggestion
#define DDS_XTypes_TRY_CONSTRUCT_TRIM (DDS_XTypes_TRY_CONSTRUCT1 | DDS_XTypes_TRY_CONSTRUCT2)
```

Reply:
You're correct that this may cause issues when combined with other operators.
I will go over the #defines in this PR and check that this is fixed.